src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/NewArrayExpression.cs

```diff
@@ -193,6 +207,7 @@ public static NewArrayExpression NewArrayBounds(Type type, params Expression[] b
         /// <param name="type">A <see cref="System.Type"/> that represents the element type of the array.</param>
         /// <param name="bounds">An <see cref="IEnumerable{T}"/> that contains <see cref="Expression"/> objects to use to populate the <see cref="NewArrayExpression.Expressions"/> collection.</param>
         /// <returns>A <see cref="NewArrayExpression"/> that has the <see cref="NodeType"/> property equal to <see cref="ExpressionType.NewArrayBounds"/> and the <see cref="NewArrayExpression.Expressions"/> property set to the specified value.</returns>
+        [RequiresDynamicCode(Expression.CallSiteRequiresDynamicCode)]
```

Comment:
I'm not sure this message will make sense to callers. This method is "NewArrayBounds", but the message will say "Creating a call site may require dynamic code generation."